src/asynqp/routing.py

```diff
@@ -122,7 +125,12 @@ def __init__(self, handler, q):
     def ready(self):
         assert not self.is_waiting, "ready() got called while waiting for a frame to be read"
         self.is_waiting = True
-        t = asyncio.async(self._read_next())
+
+        # XXX: Refactor this. There should be only 1 async task per QueueReader
+        #      It will read frames in a `while True:` loop and will be canceled
+        #      when connection is closed.
```

Comment:
This comment is accurate and germane. Presently the fact that there should be only one task running at a time is ensured by the asserts in this function and in `_read_next`. However it'd probably be cleaner and more efficient to explicitly start only one task when the `QueueReader` starts up. (I'd accept a full request fixing this.)